python/test_infra/test_utils.py

```diff
@@ -287,152 +283,78 @@ def wrapper(*args, **kwargs):
 
             assert isinstance(module, Module), (
-                f"Make sure `ttir_to_ttmetal` which was decorated with this function "
+                f"Make sure `ttir_to_ttnn` which was decorated with this function "
                 f"returns module, not file name."
             )
 
             # Convert to flatbuffer file.
-            ttmetal_to_flatbuffer_file(module, output_file_name)
+            golden_info = {}
+            for name, golden_tensor in builder.id_golden_map.items():
+                golden_info[name] = create_golden_tensor(
+                    name,
+                    list(golden_tensor.tensor.shape),
+                    list(golden_tensor.tensor.stride()),
+                    DataType.Float32,
+                    golden_tensor.tensor.data_ptr(),
+                )
+            ttnn_to_flatbuffer_file(module, output_file_name, golden_info)
 
-            print("`ttmetal_to_flatbuffer_file` passed successfully.")
+            print("`ttnn_to_flatbuffer_file` passed successfully.")
 
         return wrapper
 
     return decorator
 
 
-def translate_ttmetal_to_flatbuffer(output_file_name: str = "ttmetal_fb.ttm"):
+def ttmetal_to_flatbuffer(
+    output_file_name: str = "ttmetal_fb.ttmg",
+):
     """
-    NOTE Substitutes `ttmetal_to_flatbuffer` decorator.
-
-    By running `ttmlir-translate` on input file, it produces TTMetal flatbuffer file
-    `output_file_name`, meant to be used as a decorator on top of `ttir_to_ttmetal`
-    decorator. Take note that `ttir_to_ttmetal` has to return file name instead of
-    module if decorated with this decorator.
-
-    Wrapper around `ttmlir-translate` call.
-
-    Example
-    -------
+    Converts TTMetal module to flatbuffer and saves to file, meant to be used as a
+    decorator on top of `ttir_to_ttmetal` decorator. Take note that `ttir_to_ttmetal`
+    has to return module instead of file name if decorated with this decorator.
 
-    ```python
-    @translate_ttmetal_to_flatbuffer(output_file_name="ttmetal_fb_test_add.ttm")
-    @ttir_to_ttmetal(dump_to_file=True, output_file_name="test_add.mlir", return_module=False)
-    @compile_as_mlir_module((32, 32), (32, 32))
-    def test_add(in0: Operand, in1: Operand, builder: TTIRBuilder):
-        # CHECK: %0 = tensor.empty() : tensor<32x32xf32>
-        # CHECK: %1 = "ttir.add"(%arg0, %arg1, %0)
-        # CHECK: return %1 : tensor<32x32xf32>
-
-        return builder.add(in0, in1)
-        ```
+    Wrapper around `ttmetal_to_flatbuffer_file` pybound pass.
     """
 
-    def decorator(fn: Callable):
+    def decorator(test_fn: Callable):
         def wrapper(*args, **kwargs):
-            input_file_name = fn(*args, **kwargs)
+            # Get the TTMetal module by calling the wrapped function.
+            module = test_fn(*args, **kwargs)
 
-            assert isinstance(input_file_name, str) and os.path.isfile(
-                input_file_name
-            ), (
+            assert isinstance(module, Module), (
```

Comment:
Seems like this is obsolete since you got rid of `return_module: bool = False` from `ttir_to_ttmetal`. Update docstring above as well.

Reply:
fixed